src/components/SearchSubthoughts.tsx

```diff
@@ -1,38 +1,58 @@
-import React, { FC } from 'react'
+import React, { FC, useEffect, useState } from 'react'
 import { connect } from 'react-redux'
 import { store } from '../store'
 import { EM_TOKEN, HOME_PATH, HOME_TOKEN } from '../constants'
 import { exists } from '../selectors'
-import { searchLimit as setSearchLimit } from '../action-creators'
+import { searchContexts, searchLimit as setSearchLimit } from '../action-creators'
 import { escapeRegExp, formatNumber, isArchived, isDocumentEditable, rankThoughtsSequential, sort } from '../util'
 import Subthoughts from './Subthoughts'
 import NewThought from './NewThought'
 import { State } from '../util/initialState'
 import { Connected, Index, Lexeme, SimplePath } from '../types'
+import SearchService from '../search/searchService'
+import getFirebaseProvider from '../data-providers/firebase'
 
 interface SearchSubthoughtsProps {
   search?: string | null,
   archived?: boolean,
   searchLimit?: number,
+  remoteSearch: boolean,
   thoughtIndex: Index<Lexeme>,
 }
-
 /** Number of thoughts to limit the search results to by default. */
 const DEFAULT_SEARCH_LIMIT = 20
 
 // eslint-disable-next-line jsdoc/require-jsdoc
-const mapStateToProps = ({ archived, search, searchLimit, thoughts: { thoughtIndex } }: State) => ({
+const mapStateToProps = ({ archived, search, remoteSearch, searchLimit, thoughts: { thoughtIndex } }: State) => ({
   archived,
   search,
+  remoteSearch,
   searchLimit,
   thoughtIndex,
 })
 
 /** Subthoughts of search. */
-const SearchSubthoughts: FC<Connected<SearchSubthoughtsProps>> = ({ search, archived, searchLimit = DEFAULT_SEARCH_LIMIT, thoughtIndex, dispatch }) => {
+const SearchSubthoughts: FC<Connected<SearchSubthoughtsProps>> = ({ remoteSearch, search, archived, searchLimit = DEFAULT_SEARCH_LIMIT, thoughtIndex, dispatch }) => {
+
+  const [isRemoteSearching, setIsRemoteSearching] = useState(false)
+
+  /** Handles remote search. */
+  const handleRemoteSearch = async (value: string) => {
```

Comment:
A handler is a function that responds to an event, but this appears to be a function that performs a service request. I think `remoteSearch` or `searchRemote` would be a better name.

Reply:
Thanks. I will change it.